sound/soc/sof/ipc4-pcm.c

```diff
@@ -452,10 +452,57 @@ static int sof_ipc4_pcm_close(struct snd_soc_component *component,
 	return 0;
 }
 
+int sof_ipc4_get_time_info(struct snd_soc_component *component,
+			   struct snd_pcm_substream *substream,
+			   struct timespec64 *system_ts,
+			   struct timespec64 *audio_ts,
+			   struct snd_pcm_audio_tstamp_config *audio_tstamp_config,
+			   struct snd_pcm_audio_tstamp_report *audio_tstamp_report)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct snd_sof_pcm *spcm;
+
+	spcm = snd_sof_find_spcm_dai(component, rtd);
+	if (!spcm)
+		return -EINVAL;
+
+	if (substream->runtime->hw.info & SNDRV_PCM_INFO_HAS_LINK_ATIME &&
+	    audio_tstamp_config->type_requested == SNDRV_PCM_AUDIO_TSTAMP_TYPE_LINK) {
+		u64 nsec;
+
+		snd_pcm_gettime(substream->runtime, system_ts);
```

Comment:
you need to move this line to line 483, so that the delta between system and audio time is minimized.

Reply:
thanks